source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,797 @@
+                        receiving = pdFALSE;
+                        break;
+
+                    case kStatus_ENET_RxFrameError: /* Received an error frame.  Read & drop it */
+                        PRINTF( "RX Receive Error\n" );
+                        ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), NULL, 0, 0, NULL );
+                        /* Not sure if a trace is required.  The MAC had an error and needed to dump bytes */
+                        break;
+
+                    default:
+                        PRINTF( "RX Receive default\n" );
+                        break;
+                }
+            }
+        }
+    }
+}
+
+/**
+ * @brief Callback for ENET interrupts. We have only enabled the Ethernet receive interrupts
+ * in the case of this driver.
+ */
+static void ethernet_callback( ENET_Type * base,
+                               enet_handle_t * handle,
+                               enet_event_t event,
+                               enet_frame_info_t * frameInfo,
+                               void * userData )
+{
+    BaseType_t needsToYield = pdFALSE;
+
+    ( void ) base;
+    ( void ) handle;
+    ( void ) frameInfo;
+    ( void ) userData;
+
+    switch( event )
+    {
+        case kENET_RxEvent:
+            vTaskNotifyGiveFromISR( receiveTaskHandle, &needsToYield );
+            portEND_SWITCHING_ISR( needsToYield );
+            break;
+
+        default:
+            FreeRTOS_printf( ( "Unknown interrupt callback %u!", event ) );
+            break;
+    }
+}
+
+/**
+ * @brief This function verifies that the incoming frame needs processing.
+ *        If the frame is deemed to be appropriate, then the frame is sent to the
+ *        TCP stack for further processing.
+ * @param[in] length: The length of the incoming frame. This length should be read
+ *                    using a call to ENET_GetRxFrameSize.
+ */
+static void prvProcessFrame( int length )
+{
+    NetworkBufferDescriptor_t * pxBufferDescriptor = pxGetNetworkBufferWithDescriptor( length, 0 );
+
+    if( pxBufferDescriptor != NULL )
+    {
+        ENET_ReadFrame( ethernetifLocal->base, &( ethernetifLocal->handle ), pxBufferDescriptor->pucEthernetBuffer, length, 0, NULL );
+        pxBufferDescriptor->xDataLength = length;
+
+        if( ipCONSIDER_FRAME_FOR_PROCESSING( pxBufferDescriptor->pucEthernetBuffer ) == eProcessBuffer )
+        {
+            IPStackEvent_t xRxEvent;
+            xRxEvent.eEventType = eNetworkRxEvent;
+            xRxEvent.pvData = ( void * ) pxBufferDescriptor;
+
+            if( xSendEventStructToIPTask( &xRxEvent, 0 ) == pdFALSE )
+            {
+                vReleaseNetworkBufferAndDescriptor( pxBufferDescriptor );
+                iptraceETHERNET_RX_EVENT_LOST();
+                FreeRTOS_printf( ( "RX Event Lost\n" ) );
+            }
+        }
+        else
+        {
+            FreeRTOS_printf( ( "RX Event not to be considered\n" ) );
```

Comment:
"Dropping RX frame from xx:xx:xx:xx:xx:xx to: xx:xx:xx:xx:xx:xx, invalid target mac" or something similar would be a more useful error message here.

Reply:
Updated